compiler/analyzer.ml

```diff
@@ -103,6 +103,11 @@ let unop_error op t =
     (str_of_unop op) (str_of_type t) in
   raise (Semantic_Error message)
 
+let bool_error t = 
```

Comment:
Can we make this a general type mismatch error that takes the type as an argument? 
```ocaml
let message = sprintf "Expected type %s but got type %s instead" (str_of_type const) (str_of_type typ)
```